### PR TITLE
Notification websocket

### PR DESCRIPTION
**Pull request by Fralalala**
Added websockets for Notification.

When a price notification is created, connected users will be sent an event from the sockets to update their current notifications and notificationCount.

This PR also includes the removal of setting a token in the app.js by the sockets

### CODE REVIEW

File: server/routes/notification.js [hunk 1]
```diff
@@ -220,6 +221,24 @@ router.post("/price", async (req, res) => {
...
+    userLists.forEach((userList) => {
+      let stringUserId = `${userList.userId}`;
+
+      if (userSockets.connections[stringUserId]) {
+        if (!notifiedUsers[stringUserId]) {
+          userSockets.connections[stringUserId][0].emit("new-notifications");
+          notifiedUsers[stringUserId] = true;
+        }
+      }
+    });
```
Comment (sina-jamshidi) [inline server/routes/notification.js]: 👍

File: server/routes/notification.js [hunk 2]
```diff
@@ -220,6 +221,24 @@ router.post("/price", async (req, res) => {
...
+
+    //We iterate the userLists which contains all the affected users of the price change.
+    //We check here if the user is NOT in the notifiedUsers hashmap, if so we emit an event
+    //and add the userId to the hashmap so that if the users userId comes up again, we dont 
+    //send them another emit again.
+    userLists.forEach((userList) => {
+      let stringUserId = `${userList.userId}`;
+
+      if (userSockets.connections[stringUserId]) {
+        if (!notifiedUsers[stringUserId]) {
```
Comment (LevoraShelby) [inline server/routes/notification.js]: could use one if statement and combine the two conditions with an `&&` operator. This would be cleaner imo.

**Review by sina-jamshidi — APPROVED** [review]
looks good to me

File: server/routes/notification.js [hunk 3]
```diff
@@ -220,6 +221,24 @@ router.post("/price", async (req, res) => {
...
+          userSockets.connections[stringUserId][0].emit("new-notifications");
```
Comment (LevoraShelby) [inline server/routes/notification.js]: I would rewrite this as `userSockets.connection[stringUserId].forEach( (socket) => socket.emit("new-notification") )` so we notify the user on all instances of our site that they have open.
Also, it looks like you could send the notification info with this emit, and then we would cut down on a later backend request and db query. Just a thought though!

File: client/src/components/PriceNotification.jsx [hunk 4]
```diff
@@ -114,7 +115,7 @@ const PriceNotification = ({ notification, index }) => {
               >
                 {listLocations.length >= 2 ? (
                   <>
-                    <span className={classes.textHighlight}>{title}</span> has a
+                    <span className={classes.textHighlight}>{}</span> has a
```
Comment (LevoraShelby) [inline client/src/components/PriceNotification.jsx]: should this be `{name}` instead of `{}`?

**Review by LevoraShelby — APPROVED** [review]
Looks good. Take a look at the last comment and see if you can send the event out to all sockets a user is connected to.

**Pull request merged**